docs/decisions/0051-entity-framework-as-connector.md

```diff
@@ -0,0 +1,119 @@
+
+Sources: 
+- https://learn.microsoft.com/en-us/ef/core/miscellaneous/platforms
+- https://learn.microsoft.com/en-us/ef/efcore-and-ef6/
+
+### Existence of current SK connectors
+
+Taking into account that Semantic Kernel already has some integration with databases, which are also supported Entity Framework, there are multiple options how to proceed:
+- Support both Entity Framework and DB connector (e.g. `Microsoft.SemanticKernel.Connectors.EntityFramework` and `Microsoft.SemanticKernel.Connectors.MongoDB`) - in this case both connectors should produce exactly the same outcome, so additional work will be required (such as implementing the same set of unit/integration tests) to ensure this state. Also, any modifications to the logic should be applied in both connectors. 
+- Support just one Entity Framework connector (e.g. `Microsoft.SemanticKernel.Connectors.EntityFramework`) - in this case, existing DB connector should be removed, which may be a breaking change to existing customers. An additional work will be required to ensure that Entity Framework covers exactly the same set of features as previous DB connector.
+- Support just one DB connector (e.g. `Microsoft.SemanticKernel.Connectors.MongoDB`) - in this case, if such connector already exists - no additional work is required. If such connector doesn't exist and it's important to add it - additional work is required to implement that DB connector.
+
+
+Table with Entity Framework and Semantic Kernel database support (only for databases which support vector search):
+
+|Database Engine|Maintainer / Vendor|Supported in EF|Supported in SK|Updated to SK memory v2 design
+|-|-|-|-|-|
+|Azure Cosmos|Microsoft|Yes|Yes|Yes|
+|Azure SQL and SQL Server|Microsoft|Yes|Yes|No|
+|SQLite|Microsoft|Yes|Yes|No|
+|PostgreSQL|Npgsql Development Team|Yes|Yes|No|
+|MongoDB|MongoDB|Yes|Yes|No|
+|MySQL|Oracle|Yes|No|No|
+|Oracle DB|Oracle|Yes|No|No|
+|Google Cloud Spanner|Cloud Spanner Ecosystem|Yes|No|No|
+
+**Note**:
+One database engine can have multiple Entity Framework integrations, which can be maintained by different vendors (e.g. there are 2 MySQL EF NuGet packages - one is maintained by Oracle and another one is maintained by Pomelo Foundation Project).
+
+Vector DB connectors which are additionally supported in Semantic Kernel:
+- Azure AI Search
+- Chroma
+- Milvus
+- Pinecone
+- Qdrant
+- Redis
+- Weaviate
+
+Sources:
+- https://learn.microsoft.com/en-us/ef/core/providers/?tabs=dotnet-core-cli#current-providers
+
+## Considered Options
+
+- Add new `Microsoft.SemanticKernel.Connectors.EntityFramework` connector.
+- Do not add `Microsoft.SemanticKernel.Connectors.EntityFramework` connector, but add a new connector for individual database when needed.
+
+## Decision Outcome
+
+Based on the above investigation, the decision is not to add Entity Framework connector, but to add a new connector for individual database when needed.
```

Comment:
If I'm lazy and just want to read the outcome I'd like to get the reason why e.g.

`The reason for this decision is that Entity Framework providers do not uniformly support collection management operations and will require database specific code for key handling and object mapping. These factors will make use of an Entity Framework connector unreliable and it will not abstract away the underlying database. Additionally the number of vector databases that Entity Framework supports that Semantic Kernel does not have a memory connector for is very small.`

Reply:
Good point, thanks! Updated.